### DISCUSSION TITLE
Granular Document Notifications

### DISCUSSION THREAD

**Discussion by Raytle** [outline/outline#8133] [Ideas]
When saving a document, I would like some users to see a dialog box pop up which contains two pull down menus…

### Menu 1

- Important Update
- Ordinary Update
- Minor Update
- Trivial Update

### Menu 2

- No other users
- All other users
- User Group 1
- User Group 2
- User Group 3
- User A
- User B
- User C

A document creator could grant group(s) and/or user(s) the ability to only assign or both assign and edit choices listed under Menu 1 and/or Menu 2.

Each time a user with privileges to edit Menu 1 and/or Menu 2 tries to save a document, the user would be prompted with the two menus indicated above.

By contrast, each time a user with a privilege to edit Menu 2 (but not a privilege to edit Menu 1) tries to save a document, the user would be prompted with Menu 2, but not Menu 1.

Here’s a hypothetical use case.

Let’s say a user named Joe were to create a document named “New policy related to setting the thermostat on the new water heater.” When saving the document, Joe would be prompted to assign groups and/or users as subscribers to that document.

For example, let’s say that Joe had decides to subscribe the Maintenance Staff User Group to Important Updates for “New policy related to setting the thermostat on the new water heater” and a user named Adam to Minor Updates for the “New policy related to setting the thermostat on the new water heater.”

Any group(s) and/or user(s) subscribed to Minor Updates for a document would also receive notifications for Ordinary Updates and Important Updates to that document; however, any group/user subscribed to Important Updates to that document would also receive notifications for Important Updates to that document.

Therefore, Adam would receive a notification for all updates (except Trivial Updates) to “New policy related to setting the thermostat on the new water heater”; whereas, the Maintenance Staff User Group would only receive a notification for Important Updates to “New policy related to setting the thermostat on the new water heater.”

**Comment by tommoor** [comment]
It seems the main ask here it the ability to manage subscribers for a document, does that sound right? Right now mentioned users and those the document is shared with are automatically subscribed – so perhaps sharing with those groups / users is a reasonable approach?

**Comment by Raytle** [comment]
> It seems the main ask here it the ability to manage subscribers for a document, does that sound right? 

Yes.

> Right now mentioned users and those the document is shared with are automatically subscribed – so perhaps sharing with those groups / users is a reasonable approach?

No. That is an **un**reasonable approach. It's binary (black and white), not nuanced (various shades of grey).

I want a nuanced approach. The current approach typically results in users either receiving too many notifications (which are tantamount to spam) or too few notifications (so that users are uninformed in cases when they should be informed).

In real life, people notify other people based on all sorts of nuanced criteria. For example if a boy playing at school were to scrape his knee, his teacher probably wouldn't call the boy's parents. However, if that same boy were to break his arm, his teacher probably would call the boy's parents.

**Reply by tommoor** [reply]
Are you using an existing product that handles this well?

**Comment by Raytle** [comment]
No.

**Comment by Raytle** [comment]
Would you like a hand in shaping how these features might function?
